code/drasil-gool/GOOL/Drasil/LanguageRenderer/CppRenderer.hs

```diff
@@ -2606,17 +2606,19 @@ cpphClass :: Label -> CppHdrCode ParentSpec ->
   [CppHdrCode (StateVar CppHdrCode)] -> [CppHdrCode (Method CppHdrCode)] -> 
   CppHdrCode (Scope CppHdrCode) -> CppHdrCode (Scope CppHdrCode) -> 
   CppHdrCode (Class CppHdrCode)
-cpphClass n ps vars funcs pub priv = onCodeValue (\p -> vcat [
+cpphClass n ps vars funcs pub priv = let 
+  pubs = cpphVarsFuncsList Pub vars funcs
+  privs = cpphVarsFuncsList Priv vars funcs
+  indLi :: Doc -> Doc -> [Doc]
+  indLi pb pr
+    | isEmpty pb = [RC.scope priv <> colon, indent pr]
+    | isEmpty pr = [RC.scope pub <> colon, indent pb]
+    | otherwise   = [RC.scope pub <> colon, indent pb, blank,
+      RC.scope priv <> colon, indent privs]
```

Comment:
@muhammadaliog3 There's a helper function called `emptyIfEmpty` in Helpers.hs in the drasil-gool subpackage. I think you could use it to make this function cleaner/shorter.

Reply:
@bmaclach 
```haskell
emptyIfEmpty :: Doc -> Doc -> Doc
emptyIfEmpty ifDoc elseDoc = if isEmpty ifDoc then empty else elseDoc
```
I don't think I could use this function. Firstly because I need 3 if statements I would have to nest an ```emptyIfEmpty``` inside another ```emptyIfEmpty```, which may clutter things. The  other thing is that I need to return a list of [Doc] and not Doc. I can't just return ```pb``` if it is not empty, I need to return ```RC.scope pub <> colon``` and ```pb``` together in a list. Hence I think that using this function would complicate things.

Reply:
I was thinking you could use `emptyIfEmpty` 3 times, once for "private:", once for "public:", and once for the blank line in between. Since you'd be using it on individual elements of the list of `Doc`, the types would work. 

Having to use it three times does add some clutter, but in return you would no longer need `indLi` and the duplicated code in its 3 cases. So I think it's worth thinking about.